api/src/main/java/marquez/db/NamespaceDao.java

```diff
@@ -78,10 +78,20 @@ default Namespace upsertNamespaceMeta(
   @SqlQuery("SELECT * FROM namespaces ORDER BY name LIMIT :limit OFFSET :offset")
   List<Namespace> findAll(int limit, int offset);
 
+  @SqlUpdate("UPDATE namespaces SET is_hidden=false WHERE name = :name")
+  void undelete(String name);
+
+  @SqlUpdate("UPDATE namespaces SET is_hidden=true WHERE name = :name")
+  void delete(String name);
+
   default NamespaceRow upsertNamespaceRow(
       UUID uuid, Instant now, String name, String currentOwnerName) {
     doUpsertNamespaceRow(uuid, now, name, currentOwnerName);
-    return findNamespaceByName(name).orElseThrow();
+    NamespaceRow namespaceRow = findNamespaceByName(name).orElseThrow();
+    if (namespaceRow.getIsHidden()) {
+      undelete(namespaceRow.getName());
+    }
+    return namespaceRow;
```

Comment:
Aren't we returning a namespace row that is still hidden?

Reply:
I may be wrong but my understanding is:
 * we get `namespaceRow` which has `is_hidden=true`,
 * we perform `udelete` in database which sets `is_hidden=false`,
 * de method returns unmodified `namespaceRow` which still has wrong `is_hidden=true`

Reply:
Right, but the reference `namespaceRow` will return `namespaceRow.getIsHidden() = true` unless you query for the namespace again. Why not always set `is_hidden=true` when calling `doUpsertNamespaceRow()`? There's also `upsertNamespaceRow()` which seems to implement this reset logic? Wondering why we have both.

Reply:
The basic idea was to follow @collado-mike's comment

```

Reply:
Fixed the original issue by assigning to `namespaceRow` on `undelete`. If we want to change this logic substantially, I don't think this PR is a good place to do that.